### PR TITLE
Update: `newline-after-var` Allow comment + blank after var (fixes #2…

### LINKED ISSUE #2: Rule: Warn on unwrapped IIFEs
Should warn whenever we see this pattern:

``` js
var foo = function() {

}();
```

There should be parentheses around the function.

Rule name should be `immed`

### PR DESCRIPTION
**Pull request by IanVS**
…852)

Comments (one or more) are now allowed after a var statement in "always" mode,
as long as the comments have a blank line between them and whatever comes after.

### CODE REVIEW

File: lib/rules/newline-after-var.js [hunk 1]
```diff
@@ -99,15 +116,21 @@ module.exports = function(context) {
...
         if (mode === "never" && noNextLineToken && !hasNextLineComment) {
             context.report(node, NEVER_MESSAGE, { identifier: node.name });
         }
 
-        if (mode === "always" && (!noNextLineToken || hasNextLineComment)) {
+        if (mode === "always" && !noNextLineToken) {
             context.report(node, ALWAYS_MESSAGE, { identifier: node.name });
         }
+
+        if (mode === "always" && hasNextLineComment) {
```
Comment (BYK) [inline lib/rules/newline-after-var.js]: Looks like you can combine these:

```js
if (mode === "always"  && (!noNextLineToken || hasNextLineComment && !hasBlankLineAfterComment(nextToken, nextLineNum))) {
    context.report(node, ALWAYS_MESSAGE, { identifier: node.name });
}
```

Comment (IanVS) [inline lib/rules/newline-after-var.js]: I find that a bit tougher to read, but I will do some cleanup here...

File: lib/rules/newline-after-var.js [hunk 2]
```diff
@@ -99,13 +116,19 @@ module.exports = function(context) {
 
         // Next statement is not a `var`...
         noNextLineToken = nextToken.loc.start.line > nextLineNum;
-        hasNextLineComment = comments.indexOf(nextLineNum) >= 0;
+        hasNextLineComment = !!commentEndLine[nextLineNum];
```
Comment (nzakas) [inline lib/rules/newline-after-var.js]: What are you testing here? I'd prefer to see an explicit check rather than `!!`

Comment (IanVS) [inline lib/rules/newline-after-var.js]: I'm just testing that it's truthy (i.e. a comment exists at `nextLineNum`).  Seemed to be a cleaner check than `(typeof commentEndLine[nextLineNum] !== "undefined")`, but I can understand it's better to be explicit and will revise.

**Comment by nzakas** [comment]
LGTM

**Pull request merged**